File: packages/firebase_ml_vision/ios/Classes/SetupLocalModel.m
#import "FirebaseMlVisionPlugin.h"

@import FirebaseMLCommon;

@implementation SetupLocalModel

+ (void)modelName:(NSString *)modelName result:(FlutterResult)result {
  FIRLocalModel *localModel = [[FIRModelManager modelManager] localModelWithName:modelName];
  if (localModel == nil) {
    NSString *pathStart = @"Frameworks/App.framework/flutter_assets/assets/";
    NSString *datasetAppended = [pathStart stringByAppendingString:modelName];
    NSString *finalPath = [datasetAppended stringByAppendingString:@"/manifest.json"];
    NSString *manifestPath = [[NSBundle mainBundle] pathForResource:finalPath ofType:nil];
    FIRLocalModel *localModel = [[FIRLocalModel alloc] initWithName:modelName path:manifestPath];
    [[FIRModelManager modelManager] registerLocalModel:localModel];
    result(@"Model Setup Complete");
  } else {
    result(@"Model Already Setup");
  }
}

@end

File: packages/firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/SetupLocalModel.java
package io.flutter.plugins.firebasemlvision;

import com.google.firebase.ml.common.modeldownload.FirebaseLocalModel;
import com.google.firebase.ml.common.modeldownload.FirebaseModelManager;
import io.flutter.plugin.common.MethodChannel;

class SetupLocalModel implements Setup {
  static final SetupLocalModel instance = new SetupLocalModel();

  private SetupLocalModel() {}

  @Override
  public void setup(String modelName, MethodChannel.Result result) {
    String finalPath = "flutter_assets/assets/" + modelName + "/manifest.json";
    FirebaseLocalModel localModel = FirebaseModelManager.getInstance().getLocalModel(modelName);
    if (localModel == null) {
      localModel = new FirebaseLocalModel.Builder(modelName).setAssetFilePath(finalPath).build();
      FirebaseModelManager.getInstance().registerLocalModel(localModel);
      result.success("Model Setup Complete");
    } else {
      result.success("Model Already Setup");
    }
  }
}
